src/helpers/range_container.rs

```diff
@@ -0,0 +1,95 @@
+//! # Range Container
+
+use std::collections::BTreeMap;
+use std::ops::Range;
+
+/// type of endpoints of ranges
+pub type T = i32;
```

Comment:
it's a reasonable solution, it's just too bad we can't have multiple types of range containers :/

Reply:
so we can by templatizing it

i coded it with templates first, but there were so many traits required that i did it this way